typescript/model.ts

```diff
@@ -29,33 +48,56 @@ const metaAddressRegex = new RegExp(
 export type IMetaAddress = string;
 
 export class MetaAddress {
-    address?: string;
+    address: bigint;
     type: MetaAddressType;
-    epoch?: number;
-    addressSpace?: number;
+    epoch: bigint;
+    addressSpace: bigint;
 
-    constructor(metaAddress: string) {
-        const match = metaAddress.match(metaAddressRegex);
-        if (match !== null && match.groups !== undefined) {
-            this.type = match.groups["Type"] as MetaAddressType;
-            this.address = match.groups["Address"];
-            this.epoch = safeParseInt(match.groups["Epoch"]);
-            this.addressSpace = safeParseInt(match.groups["AddressSpace"]);
+    constructor(metaAddress: IMetaAddress);
+    constructor(address: bigint, type: MetaAddressType, epoch: bigint, addressSpace: bigint);
+    constructor(
+        stringOrAddress: string | bigint,
+        type?: MetaAddressType,
+        epoch?: bigint,
+        addressSpace?: bigint
+    ) {
+        if (typeof stringOrAddress === "string") {
+            const match = stringOrAddress.match(metaAddressRegex);
+            if (match !== null && match.groups !== undefined) {
+                this.type = match.groups["Type"] as MetaAddressType;
+                this.address = safeParseBigInt(match.groups["Address"])!;
+                this.epoch = safeParseBigInt(match.groups["Epoch"]) || 0n;
+                this.addressSpace = safeParseBigInt(match.groups["AddressSpace"]) || 0n;
+            } else {
+                this.type = "Invalid";
+            }
         } else {
-            this.type = "Invalid";
+            this.type = type || "Invalid";
+            this.address = stringOrAddress;
+            this.epoch = epoch || 0n;
+            this.addressSpace = addressSpace || 0n;
+        }
+    }
```

Comment:
Can we add validation of address, i.e., if it's too big or negative decay into `Invalid` (or throw an exception).

Reply:
Added